libs/data-mapper/src/lib/utils/Connection.Utils.ts

```diff
@@ -40,152 +47,138 @@ export const createConnectionEntryIfNeeded = (
   }
 };
 
-// Primarily used for drawn connections
-export const addNodeToConnections = (
+/**
+ * Creates or updates a connection entry's input value.
+ * This is the gateway for *setting ANY input values* in dataMapConnections - including
+ * both those that are deserialized, handle-drawn, or InputDropdown-set.
+ *
+ * @param connections The connections dictionary to add the entry to
+ * @param targetNode The node to create the connection entry for
+ * @param targetNodeReactFlowKey The reactFlowKey of `targetNode`
+ * @param inputIndex The index of the input to set the value for (only needed if NOT `isHandleDrawnOrDeserialized`)
+ * @param value The value to set the input to (InputConnection, or `null` to delete an unbounded input)
+ * @param isHandleDrawnOrDeserialized Whether the new connection being made is handle-drawn or a deserialization
+ */
+export const setConnectionInputValue = (
```

Comment:
I'll be honest, I'm not thrilled about this name. I get why you picked it and it's not inaccurate.